pkg/controller/idler/idler_controller.go

```diff
@@ -110,23 +110,18 @@ func (r *ReconcileIdler) Reconcile(request reconcile.Request) (reconcile.Result,
 			"failed to ensure idling '%s'", idler.Name)
 	}
 	// Find the earlier pod to kill and requeue. Do not requeue if no pods tracked
-	d := nextPodToBeKilledAfter(idler)
-	if d == nil || *d < time.Duration(0) {
-		// No pods tracked. Requeue after the idler timout so we don't miss new pods created within the timeout.
-		timeout := time.Duration(idler.Spec.TimeoutSeconds) * time.Second
-		d = &timeout
-	}
-	log.Info("requeueing in", "duration", d)
+	requeue, after := nextPodToBeKilledAfter(idler)
+	log.Info("requeueing in", "duration", after)
 	return reconcile.Result{
-		Requeue:      true,
-		RequeueAfter: *d,
+		Requeue:      requeue,
+		RequeueAfter: after,
```

Comment:
This is doesn't look right. We need to requeue with `RequeAfter` > 0 if there is no pods to track. In this case the requeueAfter should be equal to the idler timeout. Remember, there is no watcher to watch the pods.
This is exactly why we had the following code in the original PR:
```
	if d == nil || *d < time.Duration(0) {
		// No pods tracked. Requeue after the idler timout so we don't miss new pods created within the timeout.
		timeout := time.Duration(idler.Spec.TimeoutSeconds) * time.Second
		d = &timeout
	}
```
E2e should be failing now with this new change.

Reply:
huh? there is a watcher for the pods again: https://github.com/codeready-toolchain/member-operator/pull/205/files#diff-0ddf448ea61959821185b66811f19adbd7a7cc10c772d44722bc0cffa81a425bR63

Reply:
The initial idea was to drop pod watchers completely and just re-queue with a timeout left before the next pod to kill **or** the idler timeout (8h atm). This was still the case when I left my comment I think...
But now I see that you are restoring the watcher. We will need to test this new approach with some pods to see if there is any performance or memory usage issues.
My understanding is that you have tested the PR without watcher only. Is it correct?

Reply:
@MatousJobanek wdyt? I'm not sure what approach is better (assuming we have very similar performance/memory usage results of course). Both approaches have pros and cons.

Reply:
I like the new one more - requeue idler controller when no pod is found. It looks simpler to me:
* has only one watcher
* has no mapping (between pod and idler)
* there is always only one reconcile loop for one namespace (no less no more)
* no "error logs" for namespaces where idler resource is not present... 

But to be honest, I don't have strong opinion on it, so I don't insist on any of these solutions. I understand that the solution with the watcher may look to be more straightforward - so as you said, both approaches have pros and cons. 

I would let @xcoulon decide which solution he wants to go with since we are both fine with both of them :wink:

Reply:
Yeah I'm leaning towards the no-watcher approach too. Don't insist either though :) But still would prefer the no-watcher option...

Reply:
@alexeykazakov @MatousJobanek I've removed the usage of the pod watcher in c36a02a. When there is no pod in the namespace associated with the idler, then the controller will "requeue after" the value specified in `spec.TimeoutSeconds` to see if there are new pods that need to be tracked.

Reply:
also, if we drop the pod watcher, then we don't need https://github.com/codeready-toolchain/member-operator/pull/200 either (this latter was supposed to help filtering out reconcile loops based on pods in the `openshift-*` and `kube-*` namespaces)

Reply:
Looks good overall now but the original code also took care of the case when `nextPodToBeKilledAfter()` returns a negative duration. Kube ignores requeue requests with a negative duration. So let's add this as safety net.

Reply:
oh right, I've missed that! let me take care of it!

Reply:
https://github.com/codeready-toolchain/member-operator/pull/205#discussion_r505642209

Reply:
done in c8a7c74